src/views/bootablevolumes/actions/components/EditBootableVolumesModal.tsx

```diff
@@ -41,51 +41,106 @@ const EditBootableVolumesModal: FC<EditBootableVolumesModalProps> = ({
     () => Object.keys(convertResourceArrayToMap(preferences)).sort((a, b) => a.localeCompare(b)),
     [preferences],
   );
-  const initialMetadata: BootableVolumeMetadata = useMemo(() => {
+
+  const instanceTypesAndSizesObject = useMemo(
+    () => getInstanceTypesWithSizes(instanceTypesNames),
+    [instanceTypesNames],
+  );
+
+  const initialParams = useMemo(() => {
+    const instanceTypeLabel =
+      dataSource?.metadata?.labels?.[DEFAULT_INSTANCETYPE_LABEL]?.split('.');
+
     return {
-      labels: dataSource?.metadata?.labels,
-      annotations: dataSource?.metadata?.annotations,
+      preference: dataSource?.metadata?.labels?.[DEFAULT_PREFERENCE_LABEL],
+      instanceType: instanceTypeLabel?.[0],
+      size: instanceTypeLabel?.[1],
+      description: dataSource?.metadata?.annotations?.[ANNOTATIONS.description],
     };
   }, [dataSource]);
-  const [metadata, setMetadata] = useState<BootableVolumeMetadata>(initialMetadata);
 
-  const setBootableVolumeMetadata = (key: string, fieldKey: string) => (value: string) =>
-    setMetadata((prevState) => ({
-      ...prevState,
-      [key]: { ...prevState[key], [fieldKey]: value },
-    }));
+  const [preference, setPreference] = useState<string>(initialParams.preference);
+  const [instanceType, setInstanceType] = useState<string>(initialParams.instanceType);
+  const [size, setSize] = useState<string>(initialParams.size);
+  const [sizeOptions, setSizeOptions] = useState<string[]>(
```

Comment:
nitpick: Just for organization purposes, I would move this below the description state to keep all of the state coming from the initialParams object together. You could also use `useMemo` to have it automatically recalculate whenever `instanceType` changes instead of setting it manually.

Reply:
Okay, I'll move it below the description state. But I am not sure how to combine useState with useMemo for sizeOptions (and if it ever is a good idea)... Or how did you mean that?

Reply:
`useMemo` caches the result of a calculation and returns it on each re-render unless one of its dependencies changes. It's made specifically to watch props and state for changes.

In this case, it would look something like: 

`const sizeOptions = useMemo(() => instanceTypesToSizeMap[instanceType], [instanceType, instanceTypesToSizeMap]);`

You could then remove the call to `setSizeOptions` as it would no longer be needed.

Reply:
I understand what you mean, seems good to me, except that I am not sure if the Size drop down will be re-rendered accordingly, when value of `instanceType` changes...

Reply:
When `instanceType` changes, the component will re-render. When it re-renders, `useMemo` will know that it has changed and recompute `sizeOptions`, which will then be passed to the dropdown causing it to re-render since its props will have changed.